addons/mail/static/tests/messaging_menu/messaging_menu.test.js

```diff
@@ -921,6 +921,144 @@ test("preview should display last needaction message preview even if there is a
     });
 });
 
+test("should display attachment name with correct icon when message contains only an attachment", async () => {
+    /*
+        |-----------------------|--------------------|
+        | Attachment Type       | Icon Class         |
+        |-----------------------|--------------------|
+        | Audio message         | 'fa-microphone'    |
+        | Image attachment      | 'fa-picture-o'     |
+        | Audio attachment      | 'fa-headphones'    |
+        | Video message         | 'fa-video-camera'  |
+        | File attachment       | 'fa-file'          |
+        |-----------------------|--------------------|
+    */
+
+    const pyEnv = await startServer();
+    const partners = pyEnv["res.partner"].create(
+        Array.from({ length: 5 }, (_, i) => ({ name: `Partner${i + 1}` }))
+    );
+    const channelIds = pyEnv["discuss.channel"].create(
+        Array.from({ length: 5 }, (_, i) => ({ name: `test${i + 1}` }))
+    );
+
+    const attachments = [
+        {
+            mimetype: "audio/mpeg",
+            name: "voicemessage",
+            icon: "fa-microphone",
+            text: "Voice Message",
+            voice_ids: [Command.create({ display_name: "voicemessage" })],
+        },
+        { mimetype: "video/mp4", name: "Video.mp4", icon: "fa-video-camera", text: "Video.mp4" },
+        { mimetype: "application/pdf", name: "File.pdf", icon: "fa-file", text: "File.pdf" },
+        { mimetype: "image/jpeg", name: "Image.jpeg", icon: "fa-picture-o", text: "Image.jpeg" },
+        { mimetype: "audio/mpeg", name: "Audio.mp3", icon: "fa-headphones", text: "Audio.mp3" },
+    ];
+
+    pyEnv["mail.message"].create(
+        attachments.map((attachment, i) => ({
+            attachment_ids: [
+                Command.create({
+                    mimetype: attachment.mimetype,
+                    name: attachment.name,
+                    res_id: channelIds[i],
+                    res_model: "discuss.channel",
+                    voice_ids: attachment.voice_ids || [],
+                }),
+            ],
+            author_id: partners[i],
+            body: "",
+            model: "discuss.channel",
+            res_id: channelIds[i],
+        }))
+    );
+
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+
+    for (const [index, attachment] of attachments.entries()) {
+        await contains(
+            `.o-mail-NotificationItem:nth-child(${index + 1}) .o-mail-NotificationItem-text`,
+            { text: `Partner${index + 1}: ${attachment.text}` }
+        );
+        await contains(
+            `.o-mail-NotificationItem:nth-child(${index + 1}) .o-mail-NotificationItem-text i.${
+                attachment.icon
+            }`,
+            { count: 1 }
+        );
+    }
+});
```

Comment:
No for-loop: just dump each contains with copy-paste.
This will also remove the string interpolation, which makes the code too smart again and wastes time reading and maintaining the test.